lib/gitsh/parser.rb

```diff
@@ -115,7 +115,14 @@ def parse_and_transform(command)
     end
 
     rule(:subshell) do
-      str('$(') >> (str(')').absent? >> any).repeat(0).as(:subshell) >> str(')')
+      str('$(') >> (subshell_content).as(:subshell) >> str(')')
+    end
+
+    rule(:subshell_content) do
+      (
+        (str('(') >> subshell_content >> str(')')) |
+        (str(')').absent? >> any)
```

Comment:
This line is how you say “any single character except for `)`” in the Parslet DSL.